tools/windows-node-installer/test/e2e/aws_test.go

```diff
@@ -1,90 +1,289 @@
+	// and infrastructure IDs so that once they are created,
+	// they will be used by all subsequent testing functions.
+	createdInstance   = &ec2.Instance{}
+	createdInstanceID = ""
+	createdSgID       = ""
+	infraID           = ""
+)
+
+// SetupAWSCloudProvider creates provider ofr Cloud interface and asserts type into AWS provider.
+// This is the first step of the e2e test and function exit with fatal error if failed.
+func SetupAWSCloudProvider() *awscp.AwsProvider {
+	// The e2e test uses Microsoft Windows Server 2019 Base with Containers image, m4.large type, and libra ssh key.
 	cloud, err := cloudprovider.CloudProviderFactory(kubeconfig, awscredentials, "default", dir,
-		"ami-0b8d82dea356226d3", "m4.large", "libra")
-	assert.NoError(t, err, "error creating clients")
+		imageID, instanceType, sshKey)
+	if err != nil {
+		log.Fatalf("unable to create an instance")
+	}
+
+	// Type assert to AWS so that we can test other functionality
+	awsProvider, ok := cloud.(*awscp.AwsProvider)
+	if !ok {
+		log.Fatalf("Error asserting cloudprovider to awsProvider")
+	}
+	return awsProvider
+}
 
-	err = cloud.CreateWindowsVM()
+// TestCreateWindowsInstance creates a Windows instance on AWS and assert no error from instance creation.
+// If it fails, abandon the rest of the test and clean up.
+func TestCreateWindowsInstance(t *testing.T) {
+	err := awsProvider.CreateWindowsVM()
 	assert.NoError(t, err, "error creating Windows instance")
 	if err != nil {
 		// TODO: Remove the lingering resources as part of test setup.
-		t.Fatalf("Error while creating a VM delete the virtual machine")
-	}
-	// Type Assert to AWS so that we can test other functionality
-	aws, ok := cloud.(*awscp.AwsProvider)
-	if !ok {
-		t.Fatal("Error asserting cloudprovider to aws")
+		assert.FailNow(t, "Error while creating a VM delete the virtual machine")
 	}
+}
+
+// TestInstallerJsonFileRead reads instance and security group IDs from windows-node-installer.json file and
+// updates the createdInstanceID and CreatedSgID global variables.
+// If it fails, abandon the rest of the test and clean up.
+func TestInstallerJsonFileRead(t *testing.T) {
 	// Check instance and security group information in windows-node-installer.json.
 	info, err := resource.ReadInstallerInfo(dir + "/" + "windows-node-installer.json")
 	assert.NoError(t, err, "error reading windows-node-installer.json file")
 	assert.NotEmpty(t, info.SecurityGroupIDs, "security group is not created")
 	assert.NotEmpty(t, info.InstanceIDs, "instance is not created")
-	instance, err := aws.GetInstance(info.InstanceIDs[0])
-	if err != nil {
-		t.Fatalf("error getting ec2 instance object from instance ID %v", info.InstanceIDs[0])
+
+	if len(info.InstanceIDs) < 1 || info.InstanceIDs[0] == "" || len(info.SecurityGroupIDs) < 1 || info.
+		SecurityGroupIDs[0] == "" {
+		// TODO: Remove the lingering resources as part of test setup.
+		assert.FailNow(t, "error reading from windows-node-installer.json file")
```

Comment:
There is no error reading the file. It is just that you found the contents to be invalid.